src/compiler/checker.ts

```diff
@@ -13371,11 +13371,6 @@ namespace ts {
                         case SyntaxKind.EnumDeclaration:
                             copySymbols(getSymbolOfNode(location).exports, meaning & SymbolFlags.EnumMember);
                             break;
-                        case SyntaxKind.ClassExpression:
-                            if ((<ClassExpression>location).name) {
-                                copySymbol(location.symbol, meaning);
-                            }
-                        // Fall through

```

Comment:
You don't need it. Break from the switch and fall through in this case behaves the same